File: LEETCODE/面试题56 - I. 数组中数字出现的次数.java
class Solution {
    public int[] singleNumbers(int[] nums) {
        Map<Integer, Integer> map = new HashMap<>();
        for (int num: nums) {
            if(map.containsKey(num)) {
                map.put(num, map.get(num)+1);
            } else {
                map.put(num, 1);
            }
        }

        int[] res = new int[2];
        int i = 0;
        for(int key: map.keySet()) {
            if(map.get(key) == 1) {
                res[i] = key;
                i++;
            }
        }

        return res;
    }
}

File: LEETCODE/kotlin/面试题56 - I. 数组中数字出现的次数.kt
class Solution {
    fun singleNumbers(nums: IntArray): IntArray {
        val map: MutableMap<Int, Int> = mutableMapOf<Int, Int>()

        for (i in nums.indices) {
            var num = nums[i]
            if (map.contains(num)) {
                val cnt = map.get(num)!! + 1
                map.put(num, cnt)
            } else {
                map.put(num, 1)
            }
        }

        var res = IntArray(2)
        var i = 0
        for (key in map.keys) {
            if (map[key] == 1) {
                res[i] = key
                i++
            }
        }

        return res
    }
}